### DISCUSSION TITLE
Layout selector for output?

### DISCUSSION THREAD

**Discussion by gman0** [YaLTeR/niri#834] [Ideas]
Hello! First of all, thank you for this amazing WM!

I'm using niri on a laptop, but I'm also planning to get one of these ultra-wide screens.

```
    preset-column-widths {
        // Proportion sets the width as a fraction of the output width, taking gaps into account.
        // For example, you can perfectly fit four windows sized "proportion 0.25" on an output.
        // The default preset widths are 1/3, 1/2 and 2/3 of the output.
        proportion 0.33333
        proportion 0.5
        proportion 0.66667

        // Fixed sets the width in logical pixels exactly.
        // fixed 1920
    }
```

I'm assuming that layout config for a 16:9 screen (see above) won't fit nicely into 21:9 dimensions. Screens with more square-like aspect ratios may prefer `center-focused-column "never"`, while aspect ratios that resemble a stretched-out rectangle would probably benefit from `always`; there may be more examples like this where a globally defined layout may not be the best way to organize windows between drastically different outputs.

One way to handle these cases would be to create some sort mapping between layouts and outputs, and then let windows / columns that are created / moved-to the output behave according to the output's layout. Let me know what you think, thanks!

**Comment by YaLTeR** [comment]
Hey, #401

**Reply by gman0** [reply]
Great, that's it! Thanks!
